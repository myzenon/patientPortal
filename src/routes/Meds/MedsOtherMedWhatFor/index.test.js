import React from 'react'
import { MedsOtherMedWhatFor } from './index'
import ReactGA from 'react-ga';

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")

test.skip('renders correctly', () => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "n",
                reasonNotTaken: {} 
            } 
        },
        match: {
            params: {
                medId: '000080833' 
            }
        } 
    }
    const wrapper = render(
        <MedsOtherMedWhatFor { ...props } />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
})

test.skip('clicks NEXT button with input the reasons | api fhirCmrIdPut | go to Meds MedsAnyOtherQuestions', (done) => {
    const mockReason = "Headache"
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "n",
                reasonNotTaken: {}
            }
        },
        match: {
            params: {
                medId: '000080833'
            }
        },
        callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirCmrIdPut')
            expect(get).toEqual({
                id: mockObj.cmr.id
            })
            expect(post).toEqual({
                medication_000080833: {
                    taken: "n",
                    reasonNotTaken: {},
                    reasonCode: [{conditionName:'other', description: mockReason}],
                    reviewCompleted: true
                }
            })
            return Promise.resolve()
        },
        setCMR: (cmr) => {
            expect(cmr).toEqual({
                ...mockObj.cmr,
                medication_000080833: {
                    taken: "n",
                    reasonNotTaken: {},
                    reasonCode: [{conditionName:'other', description: mockReason}],
                    reviewCompleted: true
                }
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/meds-verify-drug/000293159')
            done()
        }
    }
    const wrapper = mount(
        <MedsAnyOtherQuestions { ...props } />
    )
    setTimeout(() => {
        wrapper.find('textarea[type="text"]').simulate('change', {
            target: {
                name: 'question',
                value: mockQuestions
            }
        })
        wrapper.find('button').simulate('click')
    }, 0)
})

test.skip('clicks NEXT button without input the questions | api fhirCmrIdPut | go to Meds 2 Verify Drug', (done) => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "n",
                reasonNotTaken: {}
            }
        },
        match: {
            params: {
                medId: '000080833'
            }
        },
        callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirCmrIdPut')
            expect(get).toEqual({
                id: mockObj.cmr.id
            })
            expect(post).toEqual({
                medication_000080833: {
                    taken: "n",
                    reasonNotTaken: {},
                    reviewCompleted: true
                }
            })
            return Promise.resolve()
        },
        setCMR: (cmr) => {
            expect(cmr).toEqual({
                ...mockObj.cmr,
                medication_000080833: {
                    taken: "n",
                    reasonNotTaken: {},
                    reviewCompleted: true
                }
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/meds-any-other-questions/otherMedication_1')
            done()
        }
    }
    const wrapper = mount(
        <MedsAnyOtherQuestions { ...props } />
    )
    setTimeout(() => {
        wrapper.find('button').simulate('click')
    }, 0)
})
