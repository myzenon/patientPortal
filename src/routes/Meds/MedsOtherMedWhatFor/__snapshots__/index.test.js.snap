// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <span
      class="heading1"
    >
      <strong>
        EFFEXOR XR 75MG
      </strong>
    </span>
    <p />
    <div>
      <div
        class="imageWrapper"
      >
        <img
          alt=""
          class="contentImg"
          src="https://rximage.nlm.nih.gov/image/images/gallery/original/00008-0833-21_RXNAVIMAGE10_3015981C.jpg"
        />
        <img
          alt="Zoom-In"
          class="imageMagnifyZoomIn"
          src="magnify-in.png"
        />
      </div>
    </div>
    <div
      class="questions"
    >
      If you would like to ask our pharmacist anything about EFFEXOR XR 75MG.
    </div>
    <p />
    <div>
      <span
        class="heading3"
      >
        Please enter it below.
      </span>
    </div>
    <p />
    <div
      class="textBoxInputWrapper"
    >
      <textarea
        autofocus=""
        name="question"
        placeholder="Write your questions here. [OPTIONAL]"
        type="text"
      />
    </div>
    <button
      class="button btn btn-secondary"
      id="btnNextMedsAnyOtherQuestions"
      type="button"
    >
      <b>
          NEXT 
      </b>
    </button>
  </div>
  <p />
  <div
    class="footerWrapper container"
  >
    <div>
      <div
        class="wrapper"
      >
        <div
          class="stepCompleted"
        >
          Account
          <br />
          Setup
          <div
            class="stepNumberActive"
          >
            1
          </div>
        </div>
        <div
          class="stepActive"
        >
          Medication
          <br />
          Review
          <div
            class="stepNumberActive"
          >
            2
          </div>
        </div>
        <div
          class="step"
        >
          Additional 
          <br />
          Medications
          <div
            class="stepNumber"
          >
            3
          </div>
        </div>
        <div
          class="step"
        >
          Health
          <br />
          Updates
          <div
            class="stepNumber"
          >
            4
          </div>
        </div>
        <div
          class="step"
        >
          Set
          <br />
          Appointment
          <div
            class="stepNumber"
          >
            5
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
